### DISCUSSION TITLE
Update perspective-viewer after server table clear

### DISCUSSION THREAD

**Discussion by mbcastledine** [perspective-dev/perspective#1797] [Q&A]
I have set a server table on a perspective-viewer instance, a `clear()` on the server side does not result in the client side clearing the data.

Is this expected behaviour? If so, is there any way for me to get the result I am after (clearing the client side) table?

**Comment by nickroci** [comment]
Think you should just be able to .remove all the keys?

**Comment by texodus** [comment]
`clear()` is not implemented for client/server replicated mode yet.  If you're using `PerspectiveWidget` in Jupyter, using server-only mode with `server = True` kwarg is a workaround, or if this is an app, you can pass your WebSocket `Table` instance directly to the viewer's `load()` method.

There are a few [open issues](https://github.com/finos/perspective/issues/1597) for this already from different contexts.
